lib/ansible/module_utils/redfish_utils.py

```diff
@@ -0,0 +1,944 @@
+# Copyright (c) 2017-2018 Dell EMC Inc.
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+import os
+import json
+import re
+import xml.etree.ElementTree as ET
+from distutils.version import LooseVersion
+from datetime import datetime
+from ansible.module_utils.urls import open_url
+from ansible.module_utils.six.moves.urllib.error import URLError, HTTPError
+
+HEADERS = {'content-type': 'application/json'}
+
+
+class RedfishUtils(object):
+
+    def __init__(self, creds, root_uri):
+        self.root_uri = root_uri
+        self.creds = creds
+        self._init_session()
+        self.default_system_id = None
+        self.default_chassis_id = None
+        self.default_manager_id = None
+        return
+
+    # The following functions are to send GET/POST/PATCH/DELETE requests
+    def get_request(self, uri):
+        if 'token' in self.creds:
+            headers = {"X-Auth-Token": self.creds['token']}
```

Comment:
Does this variable get consumed anywhere?

Reply:
token? Some OEMs require use of tokens to authenticate with their OOB controllers, so I left it here for future enhancements. But not used in this current implementation. Not opposed to removing it.

Reply:
I mean the `headers` variable. I just don't see it used anywhere else.

Reply:
The token has to be passed in headers, so it's part of the future capability thing. But then again, I'm not passing headers in open_url(), so I will remove the entire token/headers reference and document elsewhere for future implementations that require a token.